hdate/date.py

```diff
@@ -541,45 +620,202 @@ def get_omer_string(omer: int) -> str:  # pylint: disable=too-many-branches
-            else:  # days == 2
-                omer_string += "שני ימים "
+            day_mapping = {1: "יום אחד ", 2: "שני ימים "}
+            omer_string += "ו" + day_mapping.get(days, f"{ones[days]} ימים ")
+
     omer_string += "לעומר"
     return omer_string
+
+
+def _get_omer_string_english(omer: int) -> str:
+    """Représentation anglaise du compte de l'Omer."""
+    # Implémentation spécifique à l'anglais
+    ones = [
+        "",
+        "one",
+        "two",
+        "three",
+        "four",
+        "five",
+        "six",
+        "seven",
+        "eight",
+        "nine",
+    ]
+    teens = [
+        "ten",
+        "eleven",
+        "twelve",
+        "thirteen",
+        "fourteen",
+        "fifteen",
+        "sixteen",
+        "seventeen",
+        "eighteen",
+        "nineteen",
+    ]
+    tens = ["", "", "twenty", "thirty", "forty"]
+    omer_string = "Today is "
+    if omer < 10:
+        omer_string += ones[omer]
+    elif 10 <= omer < 20:
+        omer_string += teens[omer - 10]
+    else:
+        ten = omer // 10
+        one = omer % 10
+        omer_string += tens[ten]
+        if one != 0:
+            omer_string += "-" + ones[one]
+    omer_string += " day"
+    if omer != 1:
+        omer_string += "s"
+    omer_string += " of the Omer"
+    # Add weeks and days
+    weeks = omer // 7
+    days = omer % 7
+    if weeks > 0:
+        omer_string += f", which is {weeks} week"
+        if weeks != 1:
+            omer_string += "s"
+        if days > 0:
+            omer_string += f" and {days} day"
+            if days != 1:
+                omer_string += "s"
+    omer_string += "."
+    return omer_string
+
+
+def _get_omer_string_french(omer: int) -> str:
+    """Représentation française du compte de l'Omer."""
+    # Implémentation spécifique au français
+    ones = [
+        "",
+        "un",
+        "deux",
+        "trois",
+        "quatre",
+        "cinq",
+        "six",
+        "sept",
+        "huit",
+        "neuf",
+    ]
+    teens = [
+        "dix",
+        "onze",
+        "douze",
+        "treize",
+        "quatorze",
+        "quinze",
+        "seize",
+        "dix-sept",
+        "dix-huit",
+        "dix-neuf",
+    ]
+    tens = ["", "", "vingt", "trente", "quarante"]
+
+    # Dictionnaire des ordinals irréguliers
+    irregular_ordinals = {
+        1: "premier",
+        2: "deuxième",
+        3: "troisième",
+        4: "quatrième",
+        5: "cinquième",
+        6: "sixième",
+        7: "septième",
+        8: "huitième",
+        9: "neuvième",
+        10: "dixième",
+        11: "onzième",
+        12: "douzième",
+        13: "treizième",
+        14: "quatorzième",
+        15: "quinzième",
+        16: "seizième",
+        20: "vingtième",
+        30: "trentième",
+        40: "quarantième",
+    }
+
+    # Vérification que le nombre est dans la plage valide
+    if omer < 1 or omer > 49:
+        return "Le nombre doit être entre 1 et 49."
```

Comment:
No need to double check, this was checked in the general function

Reply:
ok